[lldb] Add an expression evaluation test for modules built with ...

augusto2112 opened the pull request:
-enable-testing and -enable-library-evolution

-enable-testing affects the visibility of types, which together with -enable-library-evolution may cause the compiler embedded in LLDB to generate incorrect code on expression evaluation.

augusto2112 committed:
[lldb] Add an expression evaluation test for modules built with
-enable-testing and -enable-library-evolution

-enable-testing affects the visibility of types, which together with
-enable-library-evolution may cause the compiler embedded in LLDB to
generate incorrect code on expression evaluation.

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78486
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78486
@swift-ci test

adrian-prantl commented on lldb/test/API/lang/swift/enable_testing/Makefile:
```diff
@@ -0,0 +1,22 @@
+		DYLIB_SWIFT_SOURCES:=Public.swift \
+		-f $(MAKEFILE_RULES)
+
+clean::
```
This is no longer required.

augusto2112 committed:
[lldb] Add more expression evaluation resilience tests

Add more resilience tests with a public superclass that is generic and
inherits from an objc type and a non-public subclass with a property we
access on the test. The tests vary in which modules the types exist (one
resilient executable, one resilient library and non resilient
executable, 2 resilient libraries and one resilient executable).

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78728
@swift-ci test

augusto2112 commented:
https://github.com/swiftlang/swift/pull/78728
@swift-ci test

adrian-prantl commented on lldb/test/API/lang/swift/resilience_superclass/TestSwiftResilienceSuperclass.py:
```diff
@@ -0,0 +1,15 @@
+class TestSwiftResilienceSuperclass(TestBase):
+    @skipUnlessDarwin
+    @swiftTest
+    def test(self):
```
```suggestion
    def test(self):
    """
    This test tests that ...
    """
```

adrian-prantl reviewed (commented):
Looks great, but the tests really need some comments explaining what exactly is being tested here!
